### PR TITLE
ansible-vault: add newline to the output of the 'encrypt_string' command

### LINKED ISSUE #78932: ansible-vault encrypt_string output is missing a new line, causing broken output on copy
### Summary

```
$ ansible-vault encrypt_string secret
Encryption successful
!vault |
          $ANSIBLE_VAULT;1.1;AES256
          61633966613365393435663962393261376338336136313065376437373838636336623565363239
          6630383564363634636364613830613561316333623739380a653764333864666363663539363363
          65316331636165353761626461386166633330623835316362393361343333396234663638653666
          3163373561623331340a633161373437343563333135343933376634643638613332643964313135
          3430%                                                                                                                    
 
```
If you look at the end of the printed data, it contains a `%`, which is a character that is added by the terminal in order to inform user that the executed command did not end-up with a newline.

Now you can easily guess that when user will copy/paste that yaml block, their will have a surprise, as decoding will not work,.... because of the extra `%` at the end.

This bug can easily be avoided by ensuring that we add a newline at the end of the dumped data, as this will not affect the validity of the dumped YAML, but will avoid having the terminal add the `%` character at the end of the last line, a character which will render the entire block invalid for decryption.

### Issue Type

Bug Report

### Component Name

encrypt

### Ansible Version

```console
$ ansible --version
ansible [core 2.14.0b1]
  config file = /Users/ssbarnea/.ansible.cfg
  configured module search path = ['/Users/ssbarnea/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ssbarnea/.pyenv/versions/3.11-dev/lib/python3.11/site-packages/ansible
  ansible collection location = /Users/ssbarnea/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/ssbarnea/.pyenv/versions/3.11-dev/bin/ansible
  python version = 3.11.0rc2+ (heads/3.11:8e2bda8227, Sep 19 2022, 10:59:25) [Clang 14.0.0 (clang-1400.0.29.102)] (/Users/ssbarnea/.pyenv/versions/3.11-dev/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
-
```

### OS / Environment

MacOS

### Steps to Reproduce

Just configure a ANSIBLE_VAULT_PASSWORD_FILE=.vault_pass and put a password inside the vault file, so you can easily encrypt and decypt.

Try to encrypt a secret like `ansible-vault encrypt_string secret`, select the output and paste it inside a playbook for use.

### Expected Results

Be able to copy/paste the output and be able to use it.

### Actual Results

```console
Vault format unhexlify error: Odd-length string
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by XoMute**
##### SUMMARY
Adds newline to the `ansible-vault encrypt_string ...` output.
Fixes #78932, more details here.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME

Ansible-vault: `encrypt_string` output

### CODE REVIEW

File: lib/ansible/cli/vault.py
Comment (webknjaz) [inline lib/ansible/cli/vault.py]: I think a more elegant solution would be to add one more list element and `str.join()` will do the rest:
```suggestion
        b_outs.append(b'')
        self.editor.write_data(b'\n'.join(b_outs), context.CLIARGS['output_file'] or '-')
```

**Comment by webknjaz** [comment]
@XoMute note that such changes require a change note bundled with a PR. Please add one. Here you can find additional instructions on how to do this: https://docs.ansible.com/ansible/latest/community/development_process.html#creating-changelog-fragments.

File: lib/ansible/cli/vault.py
Comment (ssbarnea) [inline lib/ansible/cli/vault.py:387]: It would a good idea to add a comment before the line, to prevent accidental removal in the future. `# Output must end with a newline to prevent bug #78932`

File: lib/ansible/cli/vault.py
Comment (XoMute) [reply lib/ansible/cli/vault.py]: @webknjaz Done

File: lib/ansible/cli/vault.py
Comment (XoMute) [reply lib/ansible/cli/vault.py:387]: @ssbarnea Done

**Comment by XoMute** [comment]
> @XoMute note that such changes require a change note bundled with a PR

I've created new changelog fragment, please review

**Comment by webknjaz** [comment]
@XoMute I've only posted the immediate concerns/prerequisites earlier, but this will need to go through the weekly Core Team triage before it can be merged. It should happen in about 3 hours. This is a call where we decide if an incoming issue or PR is desired in general, plus get multiple opinions from people with differing expertise, as opposed to having just one person decide the PR's fate.

But aside from that, LGTM except for a few minor things.

File: changelogs/fragments/79017-ansible-vault-string-encryption-ending-with-newline.yml
Comment (webknjaz) [inline changelogs/fragments/79017-ansible-vault-string-encryption-ending-with-newline.yml]: Change notes don't normally need to mention low-level implementation details. This information is useful to contributors/developers, but not the end-users. And it's already present in commits and the PR description. Changelog fragments are most useful when they explain the effect of the patch on how the end-users interface with the software, compared to how it worked in earlier versions.
How about something like this?
```suggestion
  - >-
    ansible-vault encrypt_string - started appending a line feed at the end of the encrypted string output. It was missing,
    which cased problems identifying where this string ends in some some shells (like bash) or accidentally copying an
    extra trailing terminator symbol (e.g., zsh adds prints out a ``%`` sign to signal where the original output stops)
    (https://github.com/ansible/ansible/issues/78932).
```

Comment (webknjaz) [reply changelogs/fragments/79017-ansible-vault-string-encryption-ending-with-newline.yml]: _(on an unrelated note) it's funny how a PR fixing a missing trailing LF adds a change note that does not have one :)_

Comment (XoMute) [reply changelogs/fragments/79017-ansible-vault-string-encryption-ending-with-newline.yml]: I think I saw some detailed low-level descriptions in other changelog fragments. But it reads better without it

File: lib/ansible/cli/vault.py
Comment (webknjaz) [inline lib/ansible/cli/vault.py]: I don't think that it's wise to leave the most relevant bits of the motivation in some external resource. Let's come up with something that can provide insight even offline. External references are useful for extra details.

Like this:
```suggestion
        # The output must end with a newline to play nice with terminal representation.
        # Refs:
        # * https://stackoverflow.com/a/729795/595220
        # * https://github.com/ansible/ansible/issues/78932
```

**Review by webknjaz — APPROVED** [review]

**Pull request merged**
